lib/classes/ConfigSchemaHandler/errorMessageBuilder.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+/*
+ * For error object structure, see https://github.com/epoberezkin/ajv#error-objects
+ */
+const buildErrorMessages = errors => {
+  errors = addUserFriendlyMessage(errors);
+  errors = errors.sort((a, b) => b.dataPath.length - a.dataPath.length);
+  if (!errors.length) throw new Error('Validation errors array cannot be emptry');
+  return errors[0].friendlyMessage
+    ? [errors[0].friendlyMessage]
+    : [`Error: ${errors[0].dataPath} ${errors[0].message}`];
```

Comment:
So in all cases we return just one error message?

Wouldn't it be nicer to report in a way as follows: If at least one error has `friendlyMessage`, report all errors with `friendlyMessage`. Otherwise report all errors.

Ideally I think we should expose upfront information about all errors, but of course try to not duplicate messages for same issue (hence falling back to `friendlyMessage` with ignore for not marked errors).

Reply:
> So in all cases we return just one error message?

Yes, that is correct.

> Wouldn't it be nicer to report in a way as follows: If at least one error has `friendlyMessage`, report all errors with `friendlyMessage`. Otherwise report all errors.

If we go this way, then given the example from https://github.com/serverless/serverless/pull/7335#discussion_r389819738 the user would see an error message `Unsupported function event: yourPluginEvent` even if `yourPluginEvent` is spelled correctly. If something is wrong with that event properties, then AJV will still produce an error that matches the `friendlyMessage` pattern. That is why I think it is better to show the most specific error which may not be a 'friendly' one (they are still quite understandable from what I saw).

If later we decide to use another approach, this would not be a breaking change. So I think we can leave it as it and then improve using community feedback in the next PRs.

Let me know if there are still any doubts

Reply:
Ok, thanks for explanation. Let me do some local testing, and most likely we'll follow with that

Reply:
@medikoo sure let me know how it goes